src/Node/Express/Request.purs

```diff
@@ -1,249 +1,235 @@
+import Foreign.Object (Object, lookup)
 import Node.Express.Handler (Handler, HandlerM(..))
-import Node.Express.Types (class RequestParam, ExpressM, Request, EXPRESS,
-                           Method, Protocol)
-import Node.Express.Internal.Utils (eitherToMaybe)
-import Node.Express.Internal.QueryString (Param, parse, getAll, getOne)
+import Node.Express.Types (class RequestParam, Request, Method, Protocol, decodeProtocol, decodeMethod)
 
 -- | Get route param value. If it is named route, e.g `/user/:id` then
 -- | `getRouteParam "id"` return matched part of route. If it is
 -- | regex route, e.g. `/user/(\d+)` then `getRouteParam 1` return
 -- | part that matched `(\d+)` and `getRouteParam 0` return whole
 -- | route.
-getRouteParam :: forall e a. (RequestParam a) => a -> HandlerM (express :: EXPRESS | e) (Maybe String)
+getRouteParam :: forall a. (RequestParam a) => a -> HandlerM (Maybe String)
 getRouteParam name = HandlerM \req _ _ ->
-    liftEff $ liftM1 (eitherToMaybe <<< runExcept <<< decode) (runFn2 _getRouteParam req name)
+    liftEffect $ runFn4 _getRouteParam req name Nothing Just
 
 -- | Get the request's body.
 -- | NOTE: Not parsed by default, you must attach proper middleware
 -- |       See http://expressjs.com/4x/api.html#req.body
-getBody :: forall e a. (Decode a) => HandlerM (express :: EXPRESS | e) (Either MultipleErrors a)
+getBody :: forall a. Decode a => HandlerM (F a)
 getBody = HandlerM \req _ _ ->
-    liftEff $ liftM1 (runExcept <<< decode) (_getBody req)
+    liftEffect $ liftM1 decode (_getBody req)
 
 -- | Get the request's body without a `Decode` parsing.
 -- | NOTE: Not parsed by default, you must attach proper middleware
 -- |       See http://expressjs.com/4x/api.html#req.body
-getBody' :: forall e. HandlerM (express :: EXPRESS | e) Foreign
+getBody' :: HandlerM Foreign
 getBody' = HandlerM \req _ _ ->
-    liftEff $ _getBody req
+    liftEffect $ _getBody req
 
 -- | Get param from request's body.
 -- | NOTE: Not parsed by default, you must attach proper middleware
 -- |       See http://expressjs.com/4x/api.html#req.body
-getBodyParam :: forall e a. (Decode a) => String -> HandlerM (express :: EXPRESS | e) (Maybe a)
+getBodyParam :: forall a. String -> HandlerM (Maybe a)
 getBodyParam name = HandlerM \req _ _ ->
-    liftEff $ liftM1 (eitherToMaybe <<< runExcept <<< decode) (runFn2 _getBodyParam req name)
+    liftEffect $ runFn4 _getBodyParam req name Nothing Just
 
 -- | Get param from query string (part of URL behind '?').
 -- | If there are multiple params having equal keys
 -- | return the first one.
-getQueryParam :: forall e. String -> HandlerM (express :: EXPRESS | e) (Maybe String)
+getQueryParam :: String -> HandlerM (Maybe String)
 getQueryParam name = HandlerM \req _ _ -> do
-    params <- liftEff $ queryParams req
-    pure $ getOne params name
-
--- | Get all params from query string having specified key.
-getQueryParams :: forall e. String -> HandlerM (express :: EXPRESS | e) (Array String)
```

Comment:
Could you re-implement this function as well? It is a part of public API and its removal might break somebody's setup.